src/util/gcalUtils.ts

```diff
@@ -0,0 +1,243 @@
+  return (
+    moment.duration(endTime.diff(startTime)).asHours() >= 24 ||
+    (!startTime.isSame(endTime, 'day') && endTime.hour() >= 12)
+  );
+}
+
+export function processEvents(
+  items: GCalEvent[],
+  calendarName: string
+): CalendarEvent[] {
+  const events: CalendarEvent[] = [];
+  const changed: ChangedCalendarEvent[] = [];
+  const cancelled: CancelledCalendarEvent[] = [];
+
+  items.forEach(event => {
+    if (event.originalStartTime) {
+      // cancelled or changed events
+      if (event.status === 'cancelled') {
+        // cancelled events
+        cancelled.push({
+          recurringEventId: event.recurringEventId,
+          originalStartTime: event.originalStartTime.dateTime
+            ? moment(event.originalStartTime.dateTime)
+            : moment(event.originalStartTime.date),
+        });
+      } else if (event.status === 'confirmed') {
+        // changed events
+        changed.push({
+          recurringEventId: event.recurringEventId,
+          title: event.summary,
+          description: event.description,
+          location: event.location,
+          originalStartTime: event.originalStartTime.dateTime
+            ? moment(event.originalStartTime.dateTime)
+            : moment(event.originalStartTime.date),
+          newStartTime: event.start.dateTime
+            ? moment(event.start.dateTime)
+            : moment(event.start.date),
+          newEndTime: event.end.dateTime
+            ? moment(event.end.dateTime)
+            : moment(event.end.date),
+        });
+      }
+    } else if (event.status === 'confirmed') {
+      const eventStartTime = event.start.dateTime
+        ? moment(event.start.dateTime)
+        : moment(event.start.date);
+      const eventEndTime = event.end.dateTime
+        ? moment(event.end.dateTime)
+        : moment(event.end.date);
+      // normal events
+      const newEvent: CalendarEvent = {
+        id: event.id,
+        title: event.summary,
+        startTime: eventStartTime,
+        endTime: eventEndTime,
+        description: event.description,
+        location: event.location,
+        recurrence: event.recurrence,
+        allDay: isMultiEvent(eventStartTime, eventEndTime),
+        changedEvents: [],
+        cancelledEvents: [],
+        calendarName,
+      };
+
+      events.push(newEvent);
+    }
+  });
+
+  // add changed events and cancelled events to corresponding event object
+  events.forEach(event => {
+    if (event.recurrence) {
+      changed
+        .filter(change => change.recurringEventId === event.id)
+        .forEach(change => {
+          event.changedEvents?.push(change);
+        });
+
+      cancelled
+        .filter(cancel => cancel.recurringEventId === event.id)
+        .forEach(cancel => {
+          event.cancelledEvents?.push(cancel);
+        });
+    }
+  });
+
+  return events;
+}
+
+// rrule only accepts utc time. https://github.com/jakubroztocil/rrule#important-use-utc-dates
```

Comment:
```suggestion
// rule only accepts utc time. https://github.com/jakubroztocil/rrule#important-use-utc-dates
```
nit

Reply:
This was on purpose. Rrule references the library which is short for recurrence rule